lib/src/components/buttons/button_group.dart

```diff
@@ -0,0 +1,254 @@
+
+    final BorderSide borderSide =
+        _getBorderSide(controller.value, colors, false);
+
+    return Container(
+      decoration: BoxDecoration(
+        border: Border(
+          top: borderSide,
+          left: borderSide,
+          bottom: borderSide,
+          right: (widget._isFinal) ? borderSide : BorderSide.none,
+        ),
+        borderRadius: _getRadius(borderType),
+      ),
+      padding: EdgeInsets.zero,
+      child: FilledButton(
+        statesController: controller,
+        onPressed: onPress,
+        style: getStyle(borderType, colors),
+        child: SelectionContainer.disabled(
+          child: Row(
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              if (widget.icon != null) Icon(widget.icon),
+              Text(widget.label!),
+              if (widget.onPress != null)
+                const Icon(ZetaIcons.expand_more_round),
+            ],
+          ).paddingAll(_padding),
+        ),
+      ),
+    );
+  }
+
+  double get _padding => widget.isLarge ? ZetaSpacing.x4 : ZetaSpacing.x3;
+
+  BorderSide _getBorderSide(
+      Set<MaterialState> states, ZetaColors colors, bool finalButton,) {
+    if (states.contains(MaterialState.disabled)) {
+      return BorderSide(color: colors.cool.shade40);
+    }
+    if (states.contains(MaterialState.focused)) {
+      return BorderSide(color: colors.blue, width: ZetaSpacing.x0_5);
+    }
+    return BorderSide(
+      color: finalButton ? colors.borderDefault : colors.borderSubtle,
+    );
+  }
+
+  BorderRadius _getRadius(ZetaWidgetBorder borderType) {
+    if (widget._isInitial) {
+      return borderType.radius.copyWith(
+        topRight: Radius.zero,
+        bottomRight: Radius.zero,
+      );
+    }
+    if (widget._isFinal) {
+      return borderType.radius.copyWith(
+        topLeft: Radius.zero,
+        bottomLeft: Radius.zero,
+      );
+    }
+    return ZetaRadius.none;
+  }
+
+  ButtonStyle getStyle(ZetaWidgetBorder borderType, ZetaColors colors) {
+    final ZetaColorSwatch color =
+        selected ? colors.cool : ZetaColorSwatch.fromColor(colors.black);
+
+    return ButtonStyle(
+      shape: MaterialStateProperty.all(
+        RoundedRectangleBorder(
+          borderRadius: borderType.radius,
+        ),
+      ),
+      backgroundColor: MaterialStateProperty.resolveWith<Color?>((states) {
+        if (states.contains(MaterialState.disabled)) {
+          return colors.surfaceDisabled;
+        }
+        if (states.contains(MaterialState.pressed)) {
+          return colors.primary.shade10;
+        }
+        if (states.contains(MaterialState.hovered)) {
+          return colors.cool.shade20;
+        }
+        if (selected) return colors.black;
+        return colors.white;
+      }),
+      foregroundColor: MaterialStateProperty.resolveWith<Color?>((states) {
+        if (states.contains(MaterialState.disabled)) {
+          return colors.textDisabled;
+        }
+        if (selected) return color.onColor;
```

Comment:
This should be `colors.black.onColor`

Reply:
I already set the color to be black on line 215, so will this still be necessary?

Reply:
yeah this ensure it will still work if the values of the token variables change